[DOC-10324] Added new file for function UNNEST_POS (#2930)
couchbase/docs-server: modules/n1ql/pages/n1ql-language-reference/unnestposfun.adoc

= UNNEST_POS Function
:page-topic-type: reference
:imagesdir: ../assets/images
:description: Use the UNNEST_POS() function to return the position of each element in an unnested array.

[abstract]
{description}

You can use the `UNNEST_POS()` function with the xref:n1ql-language-reference/unnest.adoc[UNNEST Clause] to return the position of each element in an unnested array. 

> [[fn-function-name,UNNEST_POS()]]

Reviewer: needs a unique anchor. Doesn't really matter here but it will when it gets moved to `metafun.adoc`.

```suggestion
[[fn-unnest-pos,UNNEST_POS()]]
```

== UNNEST_POS(`expr`)

> // Optional: only for functions with synonyms

Reviewer: Feel free to delete the template comments.

> This function has a synonym <<fn-synonym-name>>.

Reviewer: See above re: unique anchors.

```suggestion
This function has a synonym <<fn-unnest-position>>.
```

=== Description

The `UNNEST_POS` function takes an unnested array and returns the position value of each element in the array.

=== Arguments

// "None" if there are no arguments.
// Always include this section.

expr:: [Required] The alias of the unnested array from an xref:n1ql-language-reference/unnest.adoc[UNNEST Clause]. 

=== Return Values

> The `UNNEST_POS` function returns the position of each element in the unnested array, `expr`, as an integer. It returns each position value as a separate row in JSON format. 

Reviewer: Counting from 0. I know it's obvious from the example but I like to state it

In all other cases, the `UNNEST_POS` function returns `NULL` or `MISSING`.

=== Example

====
In the following example, the `UNNEST_POS` function takes the result of an `UNNEST` Clause on a given array, `a1`. The `UNNEST` function returns the position of each element in the unnested `a1` array , `u`, as the `upos` value. 

[source,N1QL]
----
include::example$functions/unnest-pos.n1ql[]
----

.Results
[source,json]
----
include::example$functions/unnest-pos.jsonc[]
----
====

> [[fn-synonym-name,UNNEST_POSITION()]]
> == UNNEST_POSITION(`expr`)
> 
> Synonym of <<fn-function-name>>.

Reviewer: See above re: unique anchors

```suggestion
[[fn-unnest-position,UNNEST_POSITION()]]
== UNNEST_POSITION(`expr`)

Synonym of <<fn-unnest-pos>>.
```
